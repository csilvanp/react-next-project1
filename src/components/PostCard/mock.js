export const postCardProps = {
  title: 'Title 1',
  body: 'Body 1',
  id: '1',
  cover: 'img/img.png'
}