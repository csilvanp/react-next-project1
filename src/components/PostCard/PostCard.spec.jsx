import { render, screen } from "@testing-library/react";
import { PostCard } from '.';
import { postCardProps } from "./mock";

const props = postCardProps;

describe('<PostCard />', () => {
  it('should render PostCard properly', () => {
    render(<PostCard {...props} />);

    expect(screen.getByRole('img', { name: /title 1/i })).toHaveAttribute('src', 'img/img.png');
    expect(screen.getByRole('heading', { name: /title 1/i })).toBeInTheDocument();
    expect(screen.getByText('Body 1')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<PostCard {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  })
})