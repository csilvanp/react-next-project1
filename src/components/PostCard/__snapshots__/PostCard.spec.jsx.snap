// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="Title 1"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h2>
      Title 1
    </h2>
    <p>
      Body 1
    </p>
  </div>
</div>
`;
